net.js

```diff
@@ -151,48 +151,55 @@ SSB.getOOO = function(msgId, cb) {
 SSB.getProfileNameAsync = function(profileId, cb) {
   // Get only the name from a profile.
   // Encapsulated differently so that implementation can be changed out for faster versions without changing the API.
-  SSB.getProfileAsync(profileId, (err, profile) => {
-    if (err)
+  console.log("Getting profile name using ssb-social-value")
+  SSB.net.about.socialValue({ key: 'name', dest: profileId }, (err, value) => {
+    if (err) {
+      console.log("Got error from ssb-social-value: " + err)
       cb(err)
-    else
-      cb(null, profile.name)
+    } else if (value) {
+      console.log("Got name from ssb-social-value: " + value)
+      cb(null, value)
+    } else {
+      // Fall back to pulling it from the server.
+      console.log("Falling back to getProfileAsync")
+      SSB.getProfileAsync(profileId, (err, profile) => {
+        if(err)
+	  cb(err)
+	else
+	  cb(null, profile.name)
+      })
+    }
   })
 }
 
 SSB.getProfileAsync = function(profileId, cb) {
-  const profiles = SSB.db.getIndex('profiles').getProfiles()
+  const keysWeWant = ['name', 'description', 'image']
+  SSB.net.about.socialValues({ keys: keysWeWant, dest: profileId }, (err, values) => {
 
-  if(profiles[profileId]) {
+  if(values) {
     // We've got it already.
-    cb(null, profiles[profileId])
+    cb(null, values)
   } else {
-    // Don't have it yet.  Let's try loading the graph.
-    SSB.db.getIndex('contacts').getGraphForFeedHops1(profileId, (err, data) => {
-      const profiles = SSB.db.getIndex('profiles').getProfiles()
-
-      if(!err && profiles[profileId]) {
-        // Got it from graph.
-        cb(null, profiles[profileId])
-      } else {
-        // Going to have to fetch it when a connection comes up.
-        SSB.connectedWithData(() => {
-          let rpc = SSB.getPeer()
-
-          if(!rpc) cb("No peer to connect to")
-
-          pull(
-            rpc.partialReplication.getMessagesOfType({id: profileId, type: 'contact'}),
-            pull.asyncMap(SSB.db.addOOO),
-            pull.collect((err, msgs) => {
-              if(err) cb(err)
-
-              const profiles = SSB.db.getIndex('profiles').getProfiles()
-              if (profiles[profileId])
-                cb(null, profiles[profileId])
-            })
-          )
+    // Going to have to fetch it when a connection comes up.
+    SSB.connectedWithData(() => {
+      let rpc = SSB.getPeer()
+
+      if(!rpc) cb("No peer to connect to")
+
+      pull(
+        rpc.partialReplication.getMessagesOfType({id: profileId, type: 'contact'}),
```

Comment:
This is a bit strange. We are fetching contact messages for the profile in question, but socialValues uses about messages.